dashboard/app/views/layouts/_header.html.haml

```diff
@@ -55,5 +55,5 @@
 - if script_level
   :javascript
     //<![CDATA[
-    dashboard.buildHeader(#{script_level.stage.summarize.to_json}, #{summarize_user_progress(script_level.script, @user || current_user).to_json}, #{script_level.level_id}, #{@user.try(:id) || 'null'}, #{@section.try(:id) || 'null'});
+    dashboard.buildHeader(#{script_level.stage.summarize.to_json}, #{current_user == nil ? 'null' : summarize_user_progress(script_level.script, @user || current_user).to_json}, #{script_level.level_id}, #{@user.try(:id) || 'null'}, #{@section.try(:id) || 'null'});
```

Comment:
The ternary expression is a bit too complex to include inline, probably better to assign the value to a variable if the ruby code after the if statement.